storage/series.go

```diff
@@ -316,21 +322,24 @@ func (s *dbSeries) Bootstrap(rs m3db.DatabaseSeriesBlocks, cutover time.Time) er
 // NB(xichen): segmentHolder is a two-item slice that's reused to
 // hold pointers to the head and the tail of each segment so we
 // don't need to allocate memory and gc it shortly after.
-func (s *dbSeries) FlushToDisk(writer m3db.FileSetWriter, blockStart time.Time, segmentHolder [][]byte) error {
+func (s *dbSeries) FlushToDisk(
+	ctx m3db.Context,
+	writer m3db.FileSetWriter,
+	blockStart time.Time,
+	segmentHolder [][]byte,
+) error {
 	s.RLock()
 	b, exists := s.blocks.GetBlockAt(blockStart)
 	if !exists {
 		s.RUnlock()
 		return nil
 	}
-	sr := b.Stream()
-	if sr == nil {
-		s.RUnlock()
-		return nil
-	}
+	sr, err := b.Stream(ctx)
 	s.RUnlock()
 
-	// TODO(xichen): register this with contexts when it's in place.
+	if err != nil || sr == nil {
+		return err
```

Comment:
Hmm, might be more explicit to `if sr == nil { return nil }` as a separate logic branch just for clarity when reading.  Take the advice or leave it 👍